pkg/controllers/clusteroperator_controller.go

```diff
@@ -68,10 +69,127 @@ func (r *ClusterOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Reques
 	if err != nil {
 		klog.Errorf("Could not determine cluster api feature gate state: %v", err)
 		return ctrl.Result{}, r.setStatusDegraded(ctx, err)
-	} else if !capiEnabled {
-		klog.Infof("FeatureGate cluster does not include cluster api. Skipping...")
-		return ctrl.Result{}, r.setStatusAvailable(ctx)
 	}
 
+	var result ctrl.Result
+	if !capiEnabled {
+		klog.Infof("FeatureGate cluster does not include cluster api. Removing...")
+		result, err = r.reconcileDelete(ctx)
+	} else {
+		klog.Infof("FeatureGate cluster does include cluster api. Installing...")
+		result, err = r.reconcile(ctx)
+	}
+	if err != nil {
+		return result, err
+	}
+
+	return result, r.setStatusAvailable(ctx)
+}
+
+func (r *ClusterOperatorReconciler) reconcile(ctx context.Context) (ctrl.Result, error) {
+	assetNames, err := assets.AssetDir("capi-operator")
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	for _, assetName := range assetNames {
+		b, err := assets.Asset(path.Join("capi-operator", assetName))
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		codecs := serializer.NewCodecFactory(r.Scheme)
+		obj, _, err := codecs.UniversalDeserializer().Decode(b, nil, nil)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if obj.GetObjectKind().GroupVersionKind().Kind == "Namespace" {
+			continue
+		}
+
+		dep, depOK := obj.(*appsv1.Deployment)
+		if depOK {
+			if err := r.customizeDeployment(dep); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		existing := obj.DeepCopyObject().(client.Object)
+		_, err = ctrl.CreateOrUpdate(ctx, r.Client, existing, func() error {
+			existing = obj.(client.Object)
+			return nil
+		})
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	// TODO wait for the deployment to be Available?
+
+	// TODO should we install
+	// - supported provider configmaps?
+	// - infrastructure secret?
+	// - provider CRs?
+
 	return ctrl.Result{}, nil
 }
+
+func (r *ClusterOperatorReconciler) customizeDeployment(dep *appsv1.Deployment) error {
+	containerToImageRef := map[string]string{
+		"manager":         "upstreamCAPIOperator",
+		"kube-rbac-proxy": "kubeRBACProxy",
+	}
+	for ci, cont := range dep.Spec.Template.Spec.Containers {
+		if imageRef, ok := containerToImageRef[cont.Name]; ok {
+			if cont.Image == r.Images[imageRef] {
+				continue
+			}
+			klog.Infof("container %s changing image from %s to %s", cont.Name, cont.Image, r.Images[imageRef])
+			dep.Spec.Template.Spec.Containers[ci].Image = r.Images[imageRef]
+		} else {
+			klog.Warningf("container %s no image replacement found for %s", cont.Name, cont.Image)
+		}
+	}
+	return setSpecHashAnnotation(&dep.ObjectMeta, dep.Spec)
+}
+
+func (r *ClusterOperatorReconciler) reconcileDelete(ctx context.Context) (ctrl.Result, error) {
```

Comment:
We are planning to support CAPI deletion after the feature gate was enabled. https://github.com/openshift/enhancements/pull/913/files#diff-a0672895c853aefb4deaafb651c091b4e24d5dc2648e340e8e85deb4de58e564R229